<launch>

  <!-- defines name of the namespace of the drone -->
  <arg name="UAV_NAME" default="MARBLE_QAV500" />
  <arg name="RUN_TYPE" default="$(env RUN_TYPE)" />

  <arg name="obj_name" default="$(env OBJ_NAME)" />
  <arg name="art_name" default="$(env ART_NAME)" />

  <arg name="standalone" default="true" />

    <!-- allows disabling the gui from command line -->
  <arg name="gui" default="true" />

    <!-- will it run using GNU debugger? -->
  <arg name="debug" default="false" />
  <arg unless="$(arg debug)" name="launch_prefix_debug" value=""/>
  <arg     if="$(arg debug)" name="launch_prefix_debug" value="debug_roslaunch"/>

    <!-- will it run as standalone nodelet or using a nodelet manager? -->
  <arg name="manager" default="$(arg UAV_NAME)_labeling_manager" />
  <arg     if="$(arg standalone)" name="nodelet" value="standalone"/>
  <arg unless="$(arg standalone)" name="nodelet" value="load"/>
  <arg     if="$(arg standalone)" name="nodelet_manager" value=""/>
  <arg unless="$(arg standalone)" name="nodelet_manager" value="$(arg manager)"/>

    <!-- Namespace -->
  <group ns="$(arg UAV_NAME)">

    <!-- EdgeDetect nodelet -->
    <node pkg="nodelet" type="nodelet" name="image_labeling_utils" args="$(arg nodelet) image_labeling_utils/ImageLabelingUtils $(arg nodelet_manager)" launch-prefix="$(arg launch_prefix_debug)" output="screen">

      <!-- Parameters loaded from launch file -->

      <param name="uav_name" type="string" value="$(arg UAV_NAME)"/>
      <param name="camera_frame_id" type="string" value="MARBLE_QAV500/base_link/camera_front_optical"/>
      <!-- <param name="dataset_name" type="string" value="randy_sim"/> -->
      <param name="dataset_name" value="$(arg art_name)"/>
      <param name="object_id" value="0"/>
      <param name="object_str" value="$(arg obj_name)"/>
      <param name="json_dir" type="string" value="$(find image_labeling_utils)/datasets/labels/"/>
      <param name="img_dir" type="string" value="$(find image_labeling_utils)/datasets/images/"/>
        <!-- ROS parameters config file -->
      <rosparam file="$(find image_labeling_utils)/config/default.yaml" />

        <!-- Subscribers -->
      <remap from="~image_in" to="front/image_raw" />
      <remap from="~camera_info_in" to="front/camera_info" />

      <remap from="~artefact_boundings" to="~artefact_boundings" />
      <remap from="~artefact_pose" to="artefact_gt" />
        <!-- Publishers -->
      <remap from="~test_publisher" to="~image_number" />

    </node>
    <node pkg="image_labeling_utils" type="artefact_gt_publisher.py" name="artefact_gt" output="screen" >
      <param name="sdf_path" value="/home/mrs/subt_workspace/src/subt_virtual/ros_packages/subt_virtual_launch/worlds/mrs_simple_cave_01.sdf"/>
      <param name="object_name" value="$(arg art_name)"/>
      <param name="obj_frame" value="MARBLE_QAV500/virtual_origin"/>
    </node>


  </group>

</launch>
